apps/condo/domains/common/utils/moneyToWords/moneyToWords.js

```diff
@@ -0,0 +1,179 @@
+const { capitalize } = require('lodash')
+
+const LOCALES = require('./locales')
+
+const FORMAT_REG_EXP = /\$([a-z]+)/gi
+
+const DEFAULT_OPTIONS = {
+    locale: 'ru',
+    currencyCode: 'RUB',
+    format: '$WholeString $wholeCurrency $decimalString $decimalCurrency',
+}
+
+/**
+ * @exception Currently only supported - 'ru' and 'en' locale 
+ * @exception Currently only supported - 'RUB' and 'USD' currencyCode 
+ * 
+ * @param { number | string } input - whole and decimal parts only through a dot. Example '124123.15'
+ * @param {{
+ *      locale: string,
+ *      currencyCode: string,
+ *      format: string,
+ * }} options 
+ * @returns { string } money converted into words
+ * @default options = {
+ *   locale: 'ru',
+ *   currencyCode: 'RUB',
+ *   format: '$WholeString $wholeCurrency $decimalString $decimalCurrency', 
+ *   }
+ * @description
+ * options.format contains the form of the string that the function will return.
+ * options.format can contain any combination of the following variables: 
+ * - $wholeString, 
+ * - $wholeCurrency, 
+ * - $WholeString, 
+ * - $WholeCurrency, 
+ * - $decimalString, 
+ * - $decimalCurrency, 
+ * - $DecimalString, 
+ * - $decimalCurrency
+ * @example moneyToWords('124', oprtions = {locale: 'en', currencyCode: 'USD', format = '$WholeString $wholeCurrency'}) returns 'One hundred twenty four dollars'
+ * @example moneyToWords('124.45', oprtions = {locale: 'en', currencyCode: 'USD', format = '$decimalString $decimalCurrency'}) returns 'fourty five cents'
+ * @example moneyToWords('123.45', oprtions = {locale: 'en', currencyCode: 'USD', format = '$WholeString $wholeCurrency $decimalString $decimalCurrency'}) returns 'One hundred twenty four dollars fourty five cents'
+ * @exception This function can work with numbers up to >10000000000. But it can be easily extended to numbers of higher orders.
+ */
+
+function moneyToWords (input, options = {}) {
+    options = { ...DEFAULT_OPTIONS, ...options }
+
+    if (!(options.locale in LOCALES)) throw new Error(`Supported only ${Object.keys(LOCALES).join(', ')} locales`)
+    if (!(options.currencyCode in LOCALES[options.locale].supportedCurrencies)) throw new Error(`Supported only ${Object.keys(LOCALES[options.locale].supportedCurrencies).join(', d')} locales`)
+
+    if (!_isValidNumber(input)) throw new Error(`Invalid Number "${input}"`)
```

Comment:
If you use function only one time and it is one liner, best to inline it. Who knows what a `validNumber` is to you?